presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -2025,11 +2025,13 @@ private static void verifyMethodHandleSignature(BoundSignature boundSignature, F
         Type returnType = boundSignature.getReturnType();
         switch (convention.getReturnConvention()) {
             case FAIL_ON_NULL:
-                verifyFunctionSignature(methodType.returnType().isAssignableFrom(returnType.getJavaType()),
+                verifyFunctionSignature(
+                        returnType.getJavaType().isAssignableFrom(methodType.returnType()),
```

Comment:
The check was previously reversed, so what is actually satisfied is _equality_.

And indeed, the stack representation is not polymorphic today, so _equality_ can be demanded.

Reply:
That's a good point.

But given how many tests fail with this change, it doesn't seem like equality is the right solution, so I'm going to take it out of the PR.